playbooks/portals-setup-following.yml

```diff
@@ -2,42 +2,71 @@
   hosts: webportals
   remote_user: "{{ webportal_user }}"
   gather_facts: True
-  serial: 1              # Limit concurrency
+  serial: 1 # Limit concurrency
   any_errors_fatal: True # Stop on first error, do not execute on the next host
-  vars:                  # Playbook specific vars
+  vars: # Playbook specific vars
     max_hosts: "{{ groups['webportals'] | length - 1 }}"
     lastpass_required: True
     portal_action: "portal-setup"
 
   tasks:
-    
     - name: Check '--limit' is used
       include_tasks: tasks/host-limit-check.yml
-    
+
     - name: Check supported OS and version
       assert:
         that:
           - ansible_distribution|lower == 'debian' and ansible_distribution_version == '10'
         fail_msg: "{{ ansible_distribution }} {{ ansible_distribution_version }} is not yet supported by this role"
 
+    # Stop docker services gracefully - begin
+    
+    # If we rerun this playbook on live/production server, we need to stop
```

Comment:
I would update the comments to say `If we rerun this playbook on any server which successfully started up services`.  Whether or not they are in production is a compounding factor but the important piece is that the server is running docker services.

Reply:
Updated.